src/runtime/proc.go

```diff
@@ -2203,6 +2203,11 @@ func execute(gp *g, inheritTime bool) {
 		traceGoStart()
 	}
 
+	// GOSB hook
+	if executeSandbox != nil {
+		executeSandbox(gp.sbid)
+	}
```

Comment:
At this point the scheduler has decided on a goroutine to execute, the hook is called just before to enable protection if necessary (gp.sbid != "" for a sandbox)